src/core/dynamicData.ts

```diff
@@ -0,0 +1,37 @@
+import { mdImageWithLink } from "./markdown";
+import { dynamicParamsUrl } from "./shieldsApi";
+
+/**
+ * Dynamic data badge.
+ *
+ * Generate a live-updating badge that references a value in a config file.
+ * Only JSON is supported.
+ */
+export function dynamicBadge(
+  label: string,
+  url: string,
+  query: string,
+  linkTarget = "",
+  altText = ""
+) {
+  if (!label) {
+    throw new Error("`label` may not be empty");
+  }
+  if (!url) {
+    throw new Error("`url` may not be empty");
+  }
+  if (!query) {
+    throw new Error("`query` may not be empty");
+  }
+  if (!altText) {
```

Comment:
Oops you forgot to use the altText functionality of `dynamicParamsUrl`

I added usage here with an input in the form.